tutorons/python/explain.py

```diff
@@ -0,0 +1,44 @@
+#! /usr/bin/env python
+# encoding: utf-8
+
+from __future__ import unicode_literals
+import logging
+from bs4 import BeautifulSoup
+import collections
+
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+
+fcns = open('tutorons/python/fcns.txt', 'r').read()
+builtin_doc = BeautifulSoup(fcns)
+dt = builtin_doc.findAll('dt')  # headers
+dd = builtin_doc.findAll('dd')  # explanations
+explanations = collections.OrderedDict()  # builtin : explanation (dt, dd)
+
+
+def resolve(a):
+    """Resolve relative hyperlinks to poin to python docs instead of current page"""
+    a['href'] = a['href'].replace(a['href'], 'https://docs.python.org/2/library/' + a['href'])
+
+# fix all relative hyperlinks
+for tag in dd:
+    map(resolve, tag.findAll('a'))
```

Comment:
Kind of weird to use `map` for the sake of producing side-effects/changing the data.  I'd recommend going with a simple, tried and true:

    for link in tag.findAll('a'):
        resolve(link)

`map` semantically usually suggests you're generating data in based on an existing list, not that you're modifying elements on the list.

Better yet, the function you're calling isn't so complex that it deserves a function, so you may just want to do:

    for tag in dd:
        for link in tag.findAll('a'):
            link['href'] = 'https://docs.python.org/2/library/' + link['href']

Nice, short, transparent.